src/kcas.ml

```diff
@@ -193,13 +192,17 @@ module Loc = struct
   let make ?(mode = Mode.obstruction_free) after =
     {
       state = Atomic.make @@ new_state after;
-      id = Id.get_unique ();
-      mode = mode;
+      id =
+        (if mode == Mode.obstruction_free then Id.get_unique () * 2
+        else (Id.get_unique () * 2) + 1);
     }
 
-  let get_id loc = loc.id [@@inline]
-  
-  let get_mode loc = loc.mode [@@inline]
+  let get_id loc = loc.id / 2 [@@inline]
+
+  let get_mode loc =
+    if loc.id mod 2 == 1 then Mode.lock_free else Mode.obstruction_free
```

Comment:
Unfortunately `x mod 2` gives a negative result if `x` is odd and negative.  `x land 1` compiles to a faster sequence of instructions.

Reply:
ohkk😅...will modify it to land.